logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt

```diff
@@ -167,17 +167,28 @@ class UserDataSource(
         }
     }
 
-    override suspend fun getKnownUser(userId: UserId) =
+    override suspend fun getKnownUser(userId: UserId): Flow<OtherUser?> =
         userDAO.getUserByQualifiedID(qualifiedID = idMapper.toDaoModel(userId))
             .map { userEntity -> userEntity?.let { publicUserMapper.fromDaoModelToPublicUser(userEntity) } }
 
+    override suspend fun observeUserInfo(userId: UserId): Flow<User?> =
+        userDAO.getUserByQualifiedID(qualifiedID = idMapper.toDaoModel(userId))
+            .map { userEntity ->
+                if (userId == getSelfUserId()) {
+                    userEntity?.let { userMapper.fromDaoModelToSelfUser(userEntity) }
+                } else {
+                    userEntity?.let { publicUserMapper.fromDaoModelToPublicUser(userEntity) }
+                }
+            }
+
```

Comment:
neat picking : this is not really returning UserInfo ? perhaps observeUser would be enough ?